src/MongoDB.Driver.Core/Core/Authentication/SaslAuthenticator.cs

```diff
@@ -163,12 +163,19 @@ private MongoAuthenticationException CreateException(IConnection connection, Exc
 
         private BsonDocument CreateStartCommand(ISaslStep currentStep)
         {
-            return new BsonDocument
+            var startCommand = new BsonDocument
             {
                 { "saslStart", 1 },
                 { "mechanism", _mechanism.Name },
                 { "payload", currentStep.BytesToSendToServer }
             };
+
+            if (_mechanism is ScramShaAuthenticator.ScramShaMechanism)
```

Comment:
If you replaced this line with:

    if (_mechanism.Name.StartsWith("SCRAM", StringComparison.OrdinalIgnoreCase))

you wouldn't have to promote anything else to internal.

Reply:
Took your suggestion!